mixin/alerts/sidecar.libsonnet

```diff
@@ -31,6 +31,7 @@
             expr: |||
               time() - max(thanos_sidecar_last_heartbeat_success_time_seconds{%(selector)s}) by (job, pod) >= 600
             ||| % thanos.sidecar,
+            'for': '2m'
```

Comment:
Hmmm we already checking if the last heartbeat is 5min ago and then with this it'll be another 2m. I think we should consider another way to prevent counter resets. WDYT?

Reply:
@kakkoyun but the first alert is using a different metric right? If you think both serve the same purpose maybe we can remove the second one?

But to me the first one is checking the liveness of the sidecar and the second one the overall health. I also do not understand the comment about 7 minutes in total. Both alerts should be evaluated in parallel. So for this particular change we just wait 2 minutes to ensure that sidecar is fully up and had the time to send a heartbeat so we do not fire for no reason.

What do you think?

Reply:
I wasn't referring to the first one. The first one checking the liveness of the Prometheus that we run `Sidecar` as sidecar. So they serve different purpose.

What I'm refer is `600` in `time() - max(thanos_sidecar_last_heartbeat_success_time_seconds{%(selector)s}) by (job, pod) >= 600`, so this triggers an alert if the last heartbeat 600 seconds ago from the current time. Hence I said 5m + 2m. 

I'm just suggesting to use something more sophisticated(?) to prevent flappy alerts by catching the counter resets.
I don't have a concrete suggestion right now I have to check docs and think about a way to achieve it.

Reply:
Oh I see. Yes in fact the delay would be 10m + 2m but I understand your point. I will look into it.

Reply:
As the `thanos_sidecar_last_heartbeat_success_time_seconds` only contains a timestamp which is updated whenever the sidecar is healthy, what do you think of using the timestamp() function? something like

`time() - max(timestamp(thanos_sidecar_last_heartbeat_success_time_seconds{%(selector)s})) by (job,pod) >= 600`

This way we will check when the metric itself was updated last instead of what value it contains as we know that it always contains a timestamp value except when it resets. Its value and its timestamp attribute they should always be the same (again, except resets).

Reply:
I have updated the code based on :point_up: